frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/services/JwtServiceTest.java

```diff
@@ -0,0 +1,111 @@
+    private static final JWTClaimsSet TEST_CLAIMS =
+            new JWTClaimsSet.Builder().claim(TEST_CLAIM_NAME, TEST_CLAIM_VALUE).build();
+    private static final JWSAlgorithm TEST_ALGORITHM = JWSAlgorithm.ES256;
+    private static final String TEST_KEY_ALIAS = "someSigningKey";
+    private static final Base64URL TEST_EXPECTED_HEADER =
+            new JWSHeader(TEST_ALGORITHM).toBase64URL();
+    private static final RSAKey TEST_KEY_PAIR_GENERATOR = generateTestKeyPair();
+    private static SignedJWT TEST_SIGNED_JWT;
+    private final KmsConnectionService kmsConnectionService = mock(KmsConnectionService.class);
+    private final JwtService jwtService = new JwtService(kmsConnectionService);
+
+    private static RSAKey generateTestKeyPair() {
+        try {
+            return new RSAKeyGenerator(2048).generate();
+        } catch (JOSEException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @BeforeEach
+    void testSetup() throws JOSEException, ParseException {
+        byte[] signatureToDER = ECDSA.transcodeSignatureToDER(TEST_SIGNATURE.decode());
+        when(kmsConnectionService.sign(any()))
+                .thenReturn(
+                        SignResponse.builder()
+                                .signature(SdkBytes.fromByteArray(signatureToDER))
+                                .build());
+        TEST_SIGNED_JWT =
+                new SignedJWT(
+                        TEST_EXPECTED_HEADER,
+                        Base64URL.encode(TEST_CLAIMS.toString()),
+                        TEST_SIGNATURE);
+    }
+
+    @Test
+    void CallsKmsToGenerateSignatureAndReturnsJWS() throws ParseException {
+        Base64URL encodedClaims = Base64URL.encode(TEST_CLAIMS.toString());
+        SdkBytes expectedMessage =
+                SdkBytes.fromByteArray(
+                        (TEST_EXPECTED_HEADER + "." + encodedClaims)
+                                .getBytes(StandardCharsets.UTF_8));
+        SignRequest expectedSignRequest =
+                SignRequest.builder()
+                        .message(expectedMessage)
+                        .keyId(TEST_KEY_ALIAS)
+                        .signingAlgorithm(SigningAlgorithmSpec.ECDSA_SHA_256)
+                        .build();
+        SignedJWT signedJWT = jwtService.signJWT(TEST_ALGORITHM, TEST_CLAIMS, TEST_KEY_ALIAS);
+        verify(kmsConnectionService).sign(expectedSignRequest);
+        assertEquals(TEST_EXPECTED_HEADER, signedJWT.getHeader().toBase64URL());
+        assertEquals(TEST_SIGNATURE.toString(), signedJWT.getSignature().toString());
+        assertEquals(TEST_CLAIM_VALUE, signedJWT.getJWTClaimsSet().getClaim(TEST_CLAIM_NAME));
+        assertEquals(encodedClaims, signedJWT.getPayload().toBase64URL());
+    }
+
+    @Test
+    void shouldEncryptJWTWithProvidedRsaKeyAndReturnJWE() throws JOSEException {
+        RSAPublicKey publicKey = TEST_KEY_PAIR_GENERATOR.toRSAPublicKey();
```

Comment:
```suggestion
        RSAPublicKey publicKey = TEST_KEY_PAIR.toRSAPublicKey();
```
is probably a better name?

Reply:
Have renamed